### PR TITLE
doc: Add Security FIPS section

### PR DESCRIPTION
**Pull request by bruno-at-bareos**
This PR add a section to explain how to setup and check FIPS enabled computer.
backport should be limited to 21 version as only newer enterprise OS are concerned.

#### Please check

- [X] Short description and the purpose of this PR is present _above this paragraph_
- [X] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 1]
```diff
@@ -93,4 +93,202 @@ Windows:
...
-\appendix
+
+.. _section-FIPS:
+
+FIPS Mode
+---------
+
+:strong:`FIPS` acronym stand for Federal Information Processing Standards.
+
+To be able to run successfully Bareos in `FIPS` mode so adjustment need to be made.
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:104]:
```suggestion
To be able to run Bareos in `FIPS` mode some adjustment need to be made.
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 2]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+
+To our knowledge only `Enterprise grade` qualified distribution can be used for the certification
+but those rules should apply also to derivative like CentOS, openSUSE.
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:107]:
```suggestion
but those rules should apply also to derivative like CentOS or openSUSE.
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 3]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+
+:strong:`beware of Signature option` all MD5 related function are disabled in `FIPS` mode.
+
+if you see error log like
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:111]:
```suggestion
if you see errors log like:
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 4]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+
+.. code-block::
+
+   Warning: MD5 digest digest initialization failed
+   Error: OpenSSL digest initialization failed: ERR=error:060800C8:digital envelope routines:EVP_DigestInit_ex:disabled for FIPS
+
+You will have to change the Signature option in you fileset to something stronger that MD5 or SHA1.
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:118]:
```suggestion
You will have to change the **Signature** option in you fileset to something stronger than MD5 or SHA1, for example **SHA256**:
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 5]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/assembly_installing-a-rhel-8-system-with-fips-mode-enabled_security-hardening
+
+The above test procedure should work out of the box, and Bareos daemons should select only FIPS approved protocols.
+
+
+SUSE Linux Enterprise 15
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+To install and activate FIPS mode under SLES you have to follow the documentation located at
+https://documentation.suse.com/sles/15-SP3/html/SLES-all/cha-security-fips.html
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:145]:
```suggestion
https://documentation.suse.com/sles/15-SP4/html/SLES-all/cha-security-fips.html
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 6]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+---------
+
+:strong:`FIPS` acronym stand for Federal Information Processing Standards.
+
+To be able to run successfully Bareos in `FIPS` mode so adjustment need to be made.
+
+To our knowledge only `Enterprise grade` qualified distribution can be used for the certification
+but those rules should apply also to derivative like CentOS, openSUSE.
+
+:strong:`beware of Signature option` all MD5 related function are disabled in `FIPS` mode.
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:109]: :```suggestion
:strong:`Beware of  the fileset *Signature* option.` All MD5 related function are disabled in `FIPS` mode.
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 7]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+
+a) Ensure fips_enabled is on
+
+.. code-block:: shell-session
+   :caption: Check FIPS enabled
+
+   sysctl -a | grep fips
+
+
+crypto.fips_enabled = 0 shows that fips is not running. If crypto.fips_enabled = 1, then fips is running.
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:198]:
```suggestion
**crypto.fips_enabled = 0** shows that fips is not running. If **crypto.fips_enabled = 1**, then fips is running.
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 8]
```diff
@@ -93,4 +93,202 @@ Windows:
...
 Our tests with the :command:`sdelete` command was not successful, as :command:`sdelete` seems to stay active in the background.
 
-\appendix
+
+.. _section-FIPS:
+
+FIPS Mode
+---------
+
+:strong:`FIPS` acronym stand for Federal Information Processing Standards.
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:102]:
```suggestion
The acronym :strong:`FIPS` stands for Federal Information Processing Standards.
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 9]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+
+Once done, openSUSE/SLE system binaries using OpenSSL will by default used only FIPS validated mechanism.
+
+
+FIPS Test procedure
+^^^^^^^^^^^^^^^^^^^
+
+An enabled `FIPS` computer can be checked with the following procedure
+
+a) Ensure fips_enabled is on
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:190]:
```suggestion
1. Ensure fips_enabled is on
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 10]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+   :caption: Check FIPS enabled
+
+   sysctl -a | grep fips
+
+
+crypto.fips_enabled = 0 shows that fips is not running. If crypto.fips_enabled = 1, then fips is running.
+to enable it refer to OS documentation (mostly adding fips=1 on boot line)
+
+
+b) Run openssl server part
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:202]:
```suggestion
2. Run openssl server part
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 11]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+
+.. code-block:: shell-session
+   :caption: run openssl server fips
+
+   OPENSSL_FORCE_FIPS_MODE=1 openssl s_server -tls1_3 -nocert -psk 1234567890
+
+
+c) Run openssl client part
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:210]:
```suggestion
3. Run openssl client part
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 12]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+
+.. code-block:: shell-session
+   :caption: run openssl client fips
+
+   OPENSSL_FORCE_FIPS_MODE=1 openssl s_client -tls1_3 -nocert -psk 1234567890
+
+
+d) Expected result, connection is established with a FIPS-140 2 compatible mechanism.
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:218]:
```suggestion
4. Expected result, connection is established with a FIPS-140 2 compatible mechanism.
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 13]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+
+   OPENSSL_FORCE_FIPS_MODE=1 openssl s_server -tls1_3 -nocert -psk 1234567890
+
+
+c) Run openssl client part
+
+.. code-block:: shell-session
+   :caption: run openssl client fips
+
+   OPENSSL_FORCE_FIPS_MODE=1 openssl s_client -tls1_3 -nocert -psk 1234567890
```
Comment (pstorz) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:215]:
```suggestion
   OPENSSL_FORCE_FIPS_MODE=1 openssl s_client -tls1_3 -psk 1234567890
```

SLE 15 SP4 does not have "-nocert" in s_client

**Review by pstorz — CHANGES_REQUESTED** [review]
I would suggest to move the "MD5 needs to be replaced by SHA256" to the end of the description. When trying to get bareos running on the fips machine, first the openssl needs to be fixed so that the daemons can even communicate, then the MD5 Problem pops up.

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst (hunk 6 above)
Comment (bruno-at-bareos) [reply docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:109]:
```suggestion
:strong:`beware of the fileset *Signature* option` all MD5 related function are disabled in `FIPS` mode.
```

File: docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst [hunk 14]
```diff
@@ -93,4 +93,202 @@ Windows:
...
+
+
+
+RedHat RHEL 8
+^^^^^^^^^^^^^
+
+For RedHat RHEL 8 follow the editor instructions located at:
+https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/assembly_installing-a-rhel-8-system-with-fips-mode-enabled_security-hardening
+
+The above test procedure should work out of the box, and Bareos daemons should select only FIPS approved protocols.
```
Comment (bruno-at-bareos) [inline docs/manuals/source/TasksAndConcepts/BareosSecurityIssues.rst:138]:
```suggestion
For RedHat RHEL 9 follow the editor instructions located at:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/security_hardening/assembly_installing-the-system-in-fips-mode_security-hardening

The above test procedure should work out of the box, and Bareos daemons should select only FIPS approved protocols.
```

**Review by pstorz — APPROVED** [review]
Looks good!

**Comment by bruno-at-bareos** [comment]
@pstorz can this one finished now and merged since approved, also check #1212 backport

**Review by pstorz — APPROVED** [review]
Good work!

**Pull request merged**